### PR TITLE
feature/COR-1857-refactor-order-of-variants

### PR DESCRIPTION
**Pull request by ben-van-eekelen**
## Summary
 
* Reordered order of variants in stacked bar chart
* Reordered order of variants in tooltip on stacked bar chart and series chart
 
#### Before
<details>
<summary>Toggle before screenshots</summary>

**Order of variants on bar chart**
image

**Order of variants in stacked bar chart tooltip**
image

**Order of variants in stacked area chart tooltip**
image

</details>
 
#### After
<details>
<summary>Toggle after screenshots</summary>

**Order of variants on bar chart**
image

**Order of variants in stacked bar chart tooltip**
image

**Order of variants in stacked area chart tooltip**
image

</details>

### CODE REVIEW

**Review by VWSCoronaDashboard30 — DISMISSED** [review]

File: packages/app/src/domain/variants/logic/reorder-and-filter.ts [hunk 1]
```diff
@@ -17,18 +17,38 @@ const hasMetricProperty = (config: any): config is { metricProperty: string } =>
...
+      1
+    )[0]
+  );
+
+  // Generate filtered tooltip context
   const reorderContext = {
     ...context,
-    config: [...context.config.filter((value) => !hasMetricProperty(value) || filteredValues[value.metricProperty] || hasSelectedMetrics)].filter(isDefined),
-    value: !hasSelectedMetrics ? filteredValues : context.value,
+    config: sortedConfigs.filter(isDefined),
```
Comment (yorickdevries) [inline packages/app/src/domain/variants/logic/reorder-and-filter.ts:50]: filter(isDefined) is redundant

File: packages/app/src/domain/variants/logic/reorder-and-filter.ts [hunk 2]
```diff
@@ -17,18 +17,38 @@ const hasMetricProperty = (config: any): config is { metricProperty: string } =>
...
+    })
+    .filter(isDefined);
+
+  // Sort variants by occurrence
+  const sortedConfigs = filteredConfigs.sort((a: any, b: any) => {
+    return context.value[b.metricProperty] - context.value[a.metricProperty];
+  });
+
+  // Move config entry 'other variants' to end
+  sortedConfigs.push(
```
Comment (yorickdevries) [inline packages/app/src/domain/variants/logic/reorder-and-filter.ts:40]: this step can be done in the sorting step (via an if statement)

File: packages/app/src/domain/variants/logic/reorder-and-filter.ts (hunk 1 above)
Comment (ben-van-eekelen) [reply packages/app/src/domain/variants/logic/reorder-and-filter.ts]: Done

**Comment by ben-van-eekelen** [comment]
Needs much more refactoring. Pick up next sprint.

**Comment by ben-van-eekelen** [comment]
TODO: Combine variant order, colour and dynamic label into 1 comprehensive metadata object for graph, table and stacked bar chart.

**Pull request merged**